src/conventional/commits.rs

```diff
@@ -92,7 +92,7 @@ impl FromStr for Commit {
                 r#"(?xms)
             ^
             (?P<type>[a-zA-Z]+)
-            (?:\((?P<scope>[a-zA-Z]+(?:-[a-zA-Z]+)*)\))?
+            (?:\((?P<scope>[[:alnum:]]+(?:-[[:alnum:]]+)*)\))?
```

Comment:
Maybe we should allow `_` and `/` as well

```suggestion
            (?:\((?P<scope>[[:alnum:]]+(?:[_/-][[:alnum:]]+)*)\))?
```

Reply:
Sounds good, how about simplifying the regex even further (and possibly allow spaces and $ as well, to match what the javascript regex is doing)?

```suggestion
            (?:\((?P<scope>[-/[:word:]]+)?\))?
```
That would then also handle an empty scope (`feat(): x` handled the same as `feat: x`)?

Reply:
Matchin the javascript regex would look like
```suggestion
            (?:\((?P<scope>[-/\$\.\ [:word:]]+)?\))?
```
I think - though looks like it wouldn't really like unicode, but that's maybe for another time..

Reply:
I think alphanumeric followed by a single slash, dash or underscore and then alphanumeric again, is the best option as it forbids bad practices for naming things.

I think the empty scope should be forbidden as well. Although the parser might be less strict for `convco changelog` and `version`, but should be stricter for `commit` and `check`. Allowing too  much now will make it difficult to change it back in the future.

Maybe there should be the possibility to specify a custom regex for the scope.

The simple version for the regex will make it possible to define silly scopes, e.g. scopes containing only spaces, or dashes.

Could you tell me which subcommands of `convco` you use?

Reply:
Ok, I've change the regex to `(?:\((?P<scope>[[:alnum:]]+(?:[-_/][[:alnum:]]+)*)\))?` in the last commit.

I'd agree that a better solution would be to have this user-configurable, given that there is no clear definition in the spec so anything even mildly strict/common-sense-y will excluding something (also, I suppose if you were to use it as a tool to enforce well-formated commit messages you most likely would want to explicitly specify which scopes are allowed). My use-case was just generating a changelog and it wasn't until I ran `check` that I realised that it wasn't including some commits.

Reply:
Thanks for the suggestion.